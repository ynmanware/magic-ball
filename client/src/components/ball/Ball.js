import React from 'react';
import './style.css';
import PropTypes from 'prop-types';

const Ball = (props) => (
    <div>
        <div className="ball">
            <div className="inner_circle">
                {props.answer ? (<div className="animated zoomIn answer"> {props.answer} </div>) : (<div className="number8"> 8 </div>)}
            </div>
        </div>
    </div>
);

Ball.propTypes = {
    answer: PropTypes.string,
};

export default Ball;