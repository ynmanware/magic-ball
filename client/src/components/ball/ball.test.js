import React from 'react';
import {shallow} from 'enzyme';
import Ball from './Ball';

describe('<Ball />', () => {
    let props;
    const createComponent = () => {
        shallow(<Ball {...props}/>);
    };

    it('should render without crashing', () => {
        props = {
            answer: 'You are lucky'
        };
        expect(createComponent).not.toThrowError();
    });

    it('should render number 8 while loading the page initially when there is not answer', () => {
        props = {};
        const wrapper = shallow(<Ball {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render supplied answer', () => {
        props = {
            answer: 'You are lucky'
        };
        const wrapper = shallow(<Ball {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });
});