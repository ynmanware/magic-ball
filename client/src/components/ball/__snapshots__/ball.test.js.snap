// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ball /> should render number 8 while loading the page initially when there is not answer 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Ball />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="ball"
      >
        <div
          className="inner_circle"
        >
          <div
            className="number8"
          >
             8 
          </div>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="inner_circle"
        >
          <div
            className="number8"
          >
             8 
          </div>
        </div>,
        "className": "ball",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="number8"
          >
             8 
          </div>,
          "className": "inner_circle",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " 8 ",
            "className": "number8",
          },
          "ref": null,
          "rendered": " 8 ",
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="ball"
        >
          <div
            className="inner_circle"
          >
            <div
              className="number8"
            >
               8 
            </div>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="inner_circle"
          >
            <div
              className="number8"
            >
               8 
            </div>
          </div>,
          "className": "ball",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="number8"
            >
               8 
            </div>,
            "className": "inner_circle",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " 8 ",
              "className": "number8",
            },
            "ref": null,
            "rendered": " 8 ",
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Ball /> should render supplied answer 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Ball
    answer="You are lucky"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="ball"
      >
        <div
          className="inner_circle"
        >
          <div
            className="animated zoomIn answer"
          >
             
            You are lucky
             
          </div>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="inner_circle"
        >
          <div
            className="animated zoomIn answer"
          >
             
            You are lucky
             
          </div>
        </div>,
        "className": "ball",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="animated zoomIn answer"
          >
             
            You are lucky
             
          </div>,
          "className": "inner_circle",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "You are lucky",
              " ",
            ],
            "className": "animated zoomIn answer",
          },
          "ref": null,
          "rendered": Array [
            " ",
            "You are lucky",
            " ",
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="ball"
        >
          <div
            className="inner_circle"
          >
            <div
              className="animated zoomIn answer"
            >
               
              You are lucky
               
            </div>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="inner_circle"
          >
            <div
              className="animated zoomIn answer"
            >
               
              You are lucky
               
            </div>
          </div>,
          "className": "ball",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="animated zoomIn answer"
            >
               
              You are lucky
               
            </div>,
            "className": "inner_circle",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "You are lucky",
                " ",
              ],
              "className": "animated zoomIn answer",
            },
            "ref": null,
            "rendered": Array [
              " ",
              "You are lucky",
              " ",
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
